pkg/api/util.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api
+
+import (
+	"context"
+	"io"
+	"time"
+)
+
+// ProgressReader tracks the progress on io.Reader Read method.
+type ProgressReader struct {
```

Comment:
I would suggest not to export types and functions unless that is required. Currently, ProgressReader and its constructor is only used in the api package. Maybe there is even a better place for ProgressReader than api package, as it is more general type, not specific to the api package.

Reply:
Agree, I exported the types and functions because of the `api_test` package. Moved to its own general package.